core/io/resource_uid.cpp

```diff
@@ -45,23 +45,44 @@ String ResourceUID::get_cache_file() {
 	return ProjectSettings::get_singleton()->get_project_data_path().path_join("uid_cache.bin");
 }
 
+static constexpr uint8_t uuid_characters[] = { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', '0', '1', '2', '3', '4', '5', '6', '7', '8' };
+static constexpr uint32_t uuid_characters_nb = (sizeof(uuid_characters) / sizeof(*(uuid_characters)));
+
 String ResourceUID::id_to_text(ID p_id) const {
 	if (p_id < 0) {
 		return "uid://<invalid>";
 	}
+	// Max 0x7FFFFFFFFFFFFFFF size is 19 digits and uid:// is 6 + 1 for null.
 	String txt;
-
+	txt.resize(26);
+
+	char32_t *p = txt.ptrw();
+	p[0] = 'u';
+	p[1] = 'i';
+	p[2] = 'd';
+	p[3] = ':';
+	p[4] = '/';
+	p[5] = '/';
+	uint32_t size = 6;
+
+	char32_t tmp[20];
+	uint32_t tmpSize = 0;
 	while (p_id) {
-		uint32_t c = p_id % base;
-		if (c < char_count) {
-			txt = String::chr('a' + c) + txt;
-		} else {
-			txt = String::chr('0' + (c - char_count)) + txt;
-		}
-		p_id /= base;
+		uint32_t c = p_id % uuid_characters_nb;
+		tmp[tmpSize] = uuid_characters[c];
+		p_id /= uuid_characters_nb;
+		++tmpSize;
 	}
 
-	return "uid://" + txt;
+	// The above loop give the number backward, recopy it in the string in the correct order.
```

Comment:
Oof, an odd problem to have. I guess it's not possible to just accept that they're reverse now?

Reply:
well no, we have to, or else it will break everything. It's not that costly anyway, that's just max 20 movs, which are not dependant, so we're probably talking around 10 cycles here.